### ISSUE TITLE
Fresh install fails with "no such file or directory"

### ISSUE THREAD

**Issue by rugaard** [laravel/valet#1188]
- Valet Version: 2.18.7
- PHP Version: 8.1.2

### Description:

When I re-installed my Mac with latest macOS `Monterey` and made a fresh install of the latest Valet version `2.18.7`, it installed fine with composer.

But when trying to run the `valet` command, it returns with the following two errors:
```shell
/Users/morten.rugaard/.composer/vendor/bin/valet: line 15: cd: ../laravel/valet: No such file or directory
/Users/morten.rugaard/.composer/vendor/bin/valet: line 37: /valet: No such file or directory
```

When going straight into the package and executing the actual valet file `/Users/morten.rugaard/.composer/vendor/laravel/valet/valet`, it works fine. _(though it asks for sudo password, just to list available commands, which I find a bit odd)_.

So basically it's the `valet` file located in `/Users/morten.rugaard/.composer/vendor/bin/valet` that is the culprit _(I believe)_.

### Steps To Reproduce:

1. Make sure Laravel Valet is completely removed from your computer.
2. Make a fresh install of latest Valet version with `composer global require laravel/valet`
3. Make sure the `$HOME/.composer/vendor/bin` is in your `$PATH`.
4. Try and execute the `valet` command in your Terminal.

### Diagnosis

```shell
# morten.rugaard in ~
❯ valet diagnose
/Users/morten.rugaard/.composer/vendor/bin/valet: line 15: cd: ../laravel/valet: No such file or directory
/Users/morten.rugaard/.composer/vendor/bin/valet: line 37: /valet: No such file or directory`
```

**Comment by driesvints** [comment]
I can't reproduce this sorry. Everything works fine for me with exact steps from above. Most likely, you're facing permission issues.

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!

**Comment by rugaard** [comment]
Thanks for the quick reply @driesvints. I have a hard time believing it's a permission issue, when other composer bin files works without a problem. I.e. the `laravel/installer` package.

The quick debugging I've done reveals that it's the `${self%[/\\]*}` of line 15 in `~/.composer/vendor/bin/valet` that returns the path `~/.composer/vendor` instead of the expected `~/.composer/vendor/bin`. Hence the `cd ../laravel/valet` part fails.

I'll try asking in the support channels as you suggests though, to see if it's actually my machine that's weird. But like I said in my issue descriptiom, my machine has just been completely re-installed and Valet was the 3rd thing I installed, so haven't done any permission stuff at any point :)

**Comment by driesvints** [comment]
@rugaard definitely odd.. I wonder if there's something wrong with your $PATH reference? But I guess not. But I managed to set everything up myself...

**Comment by Amasiani** [comment]
Had the same issue with my MacBook Air M1 but after updating composer it started working fine. Run compare global update on your terminal
